cstore_fdw.c

```diff
@@ -1740,22 +1560,21 @@ TupleCountEstimate(RelOptInfo *baserel, const char *filename)
 
 /* PageCount calculates and returns the number of pages in a file. */
 static BlockNumber
-PageCount(const char *filename)
+PageCount(Relation relation)
 {
 	BlockNumber pageCount = 0;
-	struct stat statBuffer;
+	BlockNumber footerBlockCount = 0;
+	BlockNumber dataBlockCount = 0;
 
-	/* if file doesn't exist at plan time, use default estimate for its size */
-	int statResult = stat(filename, &statBuffer);
-	if (statResult < 0)
-	{
-		statBuffer.st_size = 10 * BLCKSZ;
-	}
+	footerBlockCount = RelationGetNumberOfBlocksInFork(relation, FOOTER_FORKNUM);
+	dataBlockCount = RelationGetNumberOfBlocksInFork(relation, DATA_FORKNUM);
+
+	pageCount = footerBlockCount + dataBlockCount;
 
-	pageCount = (statBuffer.st_size + (BLCKSZ - 1)) / BLCKSZ;
-	if (pageCount < 1)
+	/* if the relation is empty, use default estimate for its size */
+	if (pageCount == 0)
 	{
-		pageCount = 1;
+		pageCount = 10;
```

Comment:
why 10?

Reply:
That's not an explanation.

Reply:
This was a very minor detail. I just wanted to preserve the existing behavior.

Removed default page count and return the actual number instead.